CONTRIBUTING.adoc

```diff
@@ -31,22 +43,36 @@ discuss it with one of our project maintainers.
 == Submission Guidelines
 
 === Submitting an Issue
-Before you submit your issue search the archive, maybe your question was already answered.
+Before you submit your issue search the archive, maybe your question was
+already answered.
 
 If your issue appears to be a bug, and hasn't been reported, open a new issue.
 Help us to maximize the effort we can spend fixing issues and adding new
-features, by not reporting duplicate issues.  Providing the following information will increase the
-chances of your issue being dealt with quickly:
+features, by not reporting duplicate issues.  Providing the following
+information will increase the chances of your issue being dealt with quickly:
 
 * **Overview of the Issue** - if an error is being thrown a stack trace helps
 * **Motivation for or Use Case** - explain why this is a bug for you
 * **Giraffle Version** - is it a named version or from our dev branch
 * **Operating System** - Mac, windows? details help
-* **Suggest a Fix** - if you can't fix the bug yourself, perhaps you can point to what might be
-  causing the problem (line of code or commit)
+* **Suggest a Fix** - if you can't fix the bug yourself, perhaps you can point
+  to what might be causing the problem (line of code or commit)
+
+
 
 === Submitting a Pull Request
 
+==== Paperwork for Pull Requests
+
+
+
+Please read this guide and make sure you agree with our {icla}.
+
+* Write good commit messages
+* Sign-off your commits `git commit --signoff` or `git commit -s`, as directed
+  by the ICL.
```

Comment:
We probably also want to add in something about how to address missing a signature.

>If you forget to sign a commit, then you'll have to do a bit of rewriting history.  Don't worry.  It's pretty easy.  If it's the latest commit, you can just run either `git commit -a -s` or `git commit --amend --signoff` to fix things.  It gets a little trickier if you have to change something farther back in history but there are some good instructions [here](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) in the Changing Multiple Commit Messages section.`